src/reducers/SupplierCreditReducer.js

```diff
@@ -36,29 +40,53 @@ export const SupplierCreditReducer = (state = initialState(), action) => {
       const { batchId, returnAmount } = payload;
 
       const rowToEditIndex = batches.findIndex(({ id }) => id === batchId);
+
+      const { totalQuantity } = batches[rowToEditIndex];
+      const adjustmentAmount = parsePositiveInteger(returnAmount);
+
       const newRow = {
         ...batches[rowToEditIndex],
-        returnAmount: parsePositiveInteger(returnAmount),
+        returnAmount: adjustmentAmount <= totalQuantity ? adjustmentAmount : 0,
       };
       batches[rowToEditIndex] = newRow;
 
       return { ...state, batches: [...batches] };
     }
 
+    case SUPPLIER_CREDIT_ACTIONS.EDIT_CATEGORY: {
+      const { payload } = action;
+      const { category } = payload;
+      return { ...state, category };
+    }
+
     case SUPPLIER_CREDIT_ACTIONS.CREATE_FROM_ITEM: {
       const { payload } = action;
       const { itemId } = payload;
 
-      const batches = UIDatabase.objects('ItemBatch').filtered(
-        'item.id == $0 && numberOfPacks > 0',
-        itemId
-      );
+      const item = UIDatabase.get('Item', itemId);
+
+      return {
+        ...state,
+        batches: mapBatchToObject(item.batches.filtered('numberOfPacks > 0')),
+        open: true,
+        item,
+        type: 'supplierCreditFromItem',
+      };
+    }
+
+    case SUPPLIER_CREDIT_ACTIONS.CREATE_FROM_INVOICE: {
+      const { payload } = action;
+      const { invoice } = payload;
+
+      const transactionBatches = invoice.getTransactionBatches(UIDatabase);
+      const itemBatches = transactionBatches.map(({ itemBatch }) => itemBatch);
```

Comment:
Since `getTransactionBatches` can return `null`, might want to do:

`const transactionBatches = invoice.getTransactionBatches(UIDatabase) ?? [];`

Or alternatively can update `Transaction.getTransactionBatches`?

Reply:
c481114737d80f378c5ca3d9b4a303eaea73e157